internal/worker/sshserver/worker.go

```diff
@@ -133,32 +133,14 @@ func (ssw *serverWrapperWorker) loop() error {
 		case <-ssw.catacomb.Dying():
 			return ssw.catacomb.ErrDying()
 		case <-changesChan:
-			// Restart the server worker.
-			srv.Kill()
-			if err := srv.Wait(); err != nil {
-				return errors.Trace(err)
-			}
-
-			// Get latest controller configuration.
-			port, _, err := ssw.getLatestControllerConfig()
-			if err != nil {
-				return errors.Trace(err)
-			}
-
-			// Start the server again.
-			srv, err = ssw.config.NewServerWorker(ServerWorkerConfig{
-				Logger:      ssw.config.Logger,
-				JumpHostKey: jumpHostKey,
-				Port:        port,
-			})
+			newPort, _, err := ssw.getLatestControllerConfig()
 			if err != nil {
-				return errors.Trace(err)
+				continue
 			}
-
-			// Re-add it to the catacomb with the newly updated configuration.
-			if err := ssw.catacomb.Add(srv); err != nil {
-				return errors.Trace(err)
+			if port == newPort {
+				continue
```

Comment:
Log something here. It won't happen so frequently that it's noisy, and experience shows it's much better to have access to this kind of logging once you get into field troubleshooting.